sphinx/directives/__init__.py

```diff
@@ -166,7 +166,7 @@ def run(self):
         node['objtype'] = node['desctype'] = self.objtype
         node['noindex'] = noindex = ('noindex' in self.options)
 
-        self.names = []  # type: List[str]
+        self.names = []  # type: List[Any]
```

Comment:
`self.names` stores the result of `handle_signature`. So this should be `Any` value.